orb-ui/cone/src/lib.rs

```diff
@@ -0,0 +1,189 @@
+pub mod button;
+pub mod lcd;
+pub mod led;
+
+use crate::button::{Button, ButtonJoinHandle};
+use crate::lcd::{Lcd, LcdCommand, LcdJoinHandle};
+use crate::led::{LedJoinHandle, LedStrip, CONE_LED_COUNT};
+use color_eyre::eyre;
+use color_eyre::eyre::Context;
+use embedded_graphics::pixelcolor::{Rgb565, RgbColor};
+use ftdi_embedded_hal::libftd2xx::{Ft4232h, Ftdi, FtdiCommon};
+use image::{ImageFormat, Luma};
+use orb_rgb::Argb;
+use std::{env, fs};
+use tokio::sync::broadcast;
+
+const CONE_FTDI_DEVICE_COUNT: usize = 8;
+
+// FIXME: FTDI adapters aren't guaranteed to enumerate in the same order every time.
+// Find a way to access a specific ftdi adapter with either a serial number of some other
+// disambiguating info.
+const CONE_FTDI_LCD_INDEX: i32 = 4;
+const CONE_FTDI_LED_INDEX: i32 = 5;
+const CONE_FTDI_BUTTON_INDEX: i32 = 7;
+
+#[derive(Debug)]
+#[allow(dead_code)]
+enum Status {
+    Connected,
+    Disconnected,
+}
+
+pub struct ConeJoinHandle {
+    lcd: LcdJoinHandle,
+    led_strip: LedJoinHandle,
+    button: ButtonJoinHandle,
+}
+
+impl ConeJoinHandle {
+    pub async fn join(self) -> eyre::Result<()> {
+        let (lcd_e, led_e, button_e) =
+            tokio::try_join!(self.lcd.0, self.led_strip.0, self.button.0)?;
+
+        // print any error that occurred
+        if let Err(e) = lcd_e {
+            tracing::error!("LCD task error: {:?}", e);
```

Comment:
Why not just return the error to the caller? Like we do here:
https://github.com/worldcoin/orb-software/blob/17e515bd1c1f91c2ac4a06bd44b9d1791f601ddf/orb-backend-state/src/main.rs#L110-L119

Reply:
I initially wanted to only print a message when a task exit with an error to debug.

I want to apply the same pattern but I can't use `map()` on `self.lcd.0` 🤔

Reply:
You need the `futures::FutureExt` [extension trait](https://rust-lang.github.io/rfcs/0445-extension-trait-conventions.html#what-is-an-extension-trait).